src/Engine/Segments/SegmentConditionModel.php

```diff
@@ -119,4 +128,40 @@ public function matchesTraitValue($traitValue): bool
 
         return $condition;
     }
+
+    private function matchesSemverTraitValue($trait, $value, $condition)
+    {
+        switch ($this->operator) {
+            case (SegmentConditions::EQUAL):
+                $condition = version_compare($trait, $value, '==');
```

Comment:
This is actually quite cool that PHP has this built in `version_compare` function.